tiledb/sm/filesystem/azure.cc

```diff
@@ -572,12 +560,19 @@ std::vector<directory_entry> Azure::ls_with_sizes(
   return entries;
 }
 
-void Azure::move_object(const URI& old_uri, const URI& new_uri) {
-  copy_blob(old_uri, new_uri);
-  remove_blob(old_uri);
+void Azure::copy_dir(const URI& old_uri, const URI& new_uri) const {
+  auto paths = ls(old_uri, "");
+  while (!paths.empty()) {
```

Comment:
Same as GCS; you should enumerate `paths` with a `for` loop and not modify them.